File: music-commons-api/src/main/java/net/transgressoft/commons/music/playlist/AudioPlaylistDirectory.java
package net.transgressoft.commons.music.playlist;

import net.transgressoft.commons.music.audio.AudioItem;

import java.util.Set;

public interface AudioPlaylistDirectory<I extends AudioItem> extends AudioPlaylist<I> {

    <N extends AudioPlaylist<I>> Set<N> descendantPlaylists();

    <N extends AudioPlaylist<I>> boolean containsPlaylist(N playlist);
}

File: music-commons-api/src/main/kotlin/net/transgressoft/commons/music/playlist/AudioPlaylistDirectory.kt
package net.transgressoft.commons.music.playlist

import net.transgressoft.commons.music.audio.AudioItem

interface AudioPlaylistDirectory<I : AudioItem, N : AudioPlaylist<I>> : AudioPlaylist<I> {

    fun descendantPlaylists(): Set<N>

    fun containsPlaylist(playlist: N): Boolean
}